server/channels/app/channels.go

```diff
@@ -84,6 +84,7 @@ type Channels struct {
 	scheduledPostMut  sync.Mutex
 	scheduledPostTask *model.ScheduledTask
 	loginAttemptsMut  sync.Mutex
+	app               *App
```

Comment:
The fact that we have to add `app` to the `Channels` struct feels wrong here. I'm trying to figure out what to propose changing.